sharding-proxy/sharding-proxy-transport/sharding-proxy-transport-postgresql/src/main/java/org/apache/shardingsphere/shardingproxy/transport/postgresql/packet/generic/PostgreSQLErrorResponsePacket.java

```diff
@@ -22,20 +22,75 @@
 import org.apache.shardingsphere.shardingproxy.transport.postgresql.packet.command.PostgreSQLCommandPacketType;
 import org.apache.shardingsphere.shardingproxy.transport.postgresql.payload.PostgreSQLPacketPayload;
 
+import java.util.HashMap;
+import java.util.Map;
+
 /**
  * Error response packet for PostgreSQL.
+ * @see <a href="https://www.postgresql.org/docs/12/protocol-message-formats.html">ErrorResponse (B)</a>
  *
  * @author zhangyonglun
+ * @author avalon566
  */
 public final class PostgreSQLErrorResponsePacket implements PostgreSQLPacket {
+
+    public static final char FIELD_TYPE_SEVERITY = 'S';
     
+    public static final char FIELD_TYPE_SEVERITY2 = 'V';
+
+    public static final char FIELD_TYPE_CODE = 'C';
+
+    public static final char FIELD_TYPE_MESSAGE = 'M';
+
+    public static final char FIELD_TYPE_DETAIL = 'D';
+
+    public static final char FIELD_TYPE_HINT = 'H';
+
+    public static final char FIELD_TYPE_POSITION = 'P';
+
+    public static final char FIELD_TYPE_INTERNAL_POSITION = 'p';
+
+    public static final char FIELD_TYPE_INTERNAL_QUERY = 'q';
+
+    public static final char FIELD_TYPE_WHERE = 'W';
+
+    public static final char FIELD_TYPE_SCHEMA_NAME = 's';
+
+    public static final char FIELD_TYPE_TABLE_NAME = 't';
+
+    public static final char FIELD_TYPE_COLUMN_NAME = 'c';
+
+    public static final char FIELD_TYPE_DATA_TYPE_NAME = 'd';
+
+    public static final char FIELD_TYPE_CONSTRAINT_NAME = 'n';
+
+    public static final char FIELD_TYPE_FILE = 'F';
+
+    public static final char FIELD_TYPE_LINE = 'L';
+
+    public static final char FIELD_TYPE_ROUTINE = 'R';
+
     @Getter
     private final char messageType = PostgreSQLCommandPacketType.ERROR_RESPONSE.getValue();
-    
-    private final char errorCode = '\0';
-    
+
+    private final Map<Character, String> fields = new HashMap<>();
+
     @Override
     public void write(final PostgreSQLPacketPayload payload) {
-        payload.writeInt1(errorCode);
+        for (Map.Entry<Character, String> each : fields.entrySet()) {
```

Comment:
Sorry, I didn't notice this before. Please:
import java.util.Map.Entry;
and
Map.Entry -> Entry